packages/Interface/src/DTK_C_API.cpp

```diff
@@ -349,98 +363,135 @@ bool DTK_is_initialized() { return DataTransferKit::isInitialized(); }
 
 void DTK_finalize() { DataTransferKit::finalize(); }
 
-void DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
-                       void ( *f )(), void *user_data )
+int DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
+                      void ( *f )(), void *user_data )
```

Comment:
I was trying to figure out why `strerror` returns `char *` and not `const char *`. The answer is [quite interesting](https://stackoverflow.com/questions/3096237/why-doesnt-strerror-return-a-const-qualified-pointer):
```
It was already like that in C89, describing a practice dating back to
before the introduction of const in the language.
```